### PR TITLE
Add instruction how to use env and build args

### LINKED ISSUE #12: Mention TARANTOOL_SERVER_USERNAME and other env options in README
subj

### PR DESCRIPTION
**Pull request by nickkkccc**
I haven't forgotten about:
- [x] Tests
- [ ] Changelog
- [ ] Documentation
- [x] Commit messages comply with the [guideline](https://www.tarantool.io/en/doc/latest/dev_guide/developer_guidelines/#how-to-write-a-commit-message)
- [x] Cleanup the code for review. See [checklist](https://github.com/tarantool/cartridge-java/blob/master/docs/review-checklist.md)

Related issues:
Closes #12

### CODE REVIEW

**Review by ArtDu — CHANGES_REQUESTED** [review]

File: README.md [hunk 1]
```diff
@@ -167,16 +167,20 @@ public class SomeOtherTest {
...
-            .withDirectoryBinding("cartridge")
-            .withRouterHost("localhost") // Optional, "localhost" is default
-            .withRouterPort(3301) // Binary port, optional, 3301 is default
-            .withAPIPort(8801) // Cartridge HTTP API port, optional, 8081 is default
-            .withRouterUsername("admin") // Specify the actual username, default is "admin"
-            .withRouterPassword("testapp-cluster-cookie") // Specify the actual password, see the "cluster_cookie" parameter
-                                                          // in the cartridge.cfg({...}) call in your application.
-                                                          // Usually it can be found in the init.lua module
-            .withReuse(true); // allows to reuse the container build once for faster testing
+            // Optional, "localhost" is default
```
[the remark this answers is not in the document]
Comment (nickkkccc) [reply README.md]: Fixed

File: README.md [hunk 2]
```diff
@@ -167,16 +167,20 @@ public class SomeOtherTest {
...
+            .withRouterHost("localhost")
+            // Binary port, optional, 3301 is default
```
[the remark this answers is not in the document]
Comment (nickkkccc) [reply README.md]: Fixed

File: README.md [hunk 3]
```diff
@@ -193,6 +197,185 @@ public class SomeOtherTest {
     }
 ```
 
+##### Environment variables of cartridge container and build arguments:
+###### Build arguments:
+
+This section describes the Docker image build arguments and environment variables inside the container. It is worth 
+noting that all build arguments listed here are passed into environment variables of the same name. At the moment,
```
[the remark this answers is not in the document]
Comment (nickkkccc) [reply README.md]: Fixed

File: README.md [hunk 4]
```diff
@@ -193,6 +197,185 @@ public class SomeOtherTest {
...
 
+##### Environment variables of cartridge container and build arguments:
+###### Build arguments:
+
+This section describes the Docker image build arguments and environment variables inside the container. It is worth 
+noting that all build arguments listed here are passed into environment variables of the same name. At the moment, 
+the following arguments are available to build the image:
+
+- `CARTRIDGE_SRC_DIR` - directory on the host machine that contains all the .lua scripts needed to initialize and run 
+cartridge. Defaults to `cartridge`. Does not convert to an environment variable.
```
[the remark this answers is not in the document]
Comment (nickkkccc) [reply README.md]: Fixed

File: README.md [hunk 5]
```diff
@@ -193,6 +197,185 @@ public class SomeOtherTest {
...
+- `TARANTOOL_WORKDIR` - a directory where all data will be stored: snapshots, wal logs and cartridge config file.
```
[the remark this answers is not in the document]
Comment (nickkkccc) [reply README.md]: Fixed

File: README.md [hunk 6]
```diff
@@ -193,6 +197,185 @@ public class SomeOtherTest {
...
+###### Build arguments:
+
+This section describes the Docker image build arguments and environment variables inside the container. It is worth 
+noting that all build arguments listed here are passed into environment variables of the same name. At the moment, 
+the following arguments are available to build the image:
+
+- `CARTRIDGE_SRC_DIR` - directory on the host machine that contains all the .lua scripts needed to initialize and run 
+cartridge. Defaults to `cartridge`. Does not convert to an environment variable.
+- `TARANTOOL_WORKDIR` - a directory where all data will be stored: snapshots, wal logs and cartridge config file. 
+Defaults to `/app`. Converts to an environment variable. It is not recommended to override via the `withEnv(...)` method.
```
[the remark this answers is not in the document]
Comment (nickkkccc) [reply README.md]: Fixed

File: README.md [hunk 7]
```diff
@@ -193,6 +197,185 @@ public class SomeOtherTest {
...
+- variable. It is not recommended to override via the `withEnv(...)` method.
+- `TARANTOOL_INSTANCES_FILE` - path to the configuration file. Defaults to `./instances.yml`. Converts to an environment
+variable. It is not recommended to override via the `withEnv(...)` method.
+- `START_DELAY` - the time after which cartridge will actually run after the container has started. Converts to an 
+environment variable. It is not recommended to override via the `withEnv(...)` method.
+
+You can set the Docker image build arguments using a map, which is passed as an input argument to the constructor when
+creating a container in Java code:
+
+```java
```
[the remark this answers is not in the document]
Comment (nickkkccc) [reply README.md]: Fixed

File: src/test/java/org/testcontainers/containers/CartridgeContainerTestUtils.java [hunk 8]
```diff
@@ -21,4 +24,18 @@ static public void executeProfileReplaceSmokeTest(TarantoolCartridgeContainer co
         assertEquals(1, result.size());
         assertEquals(33, ((List<?>) result.get(0)).get(3));
     }
+
+    public static boolean envIsContainsInStdout(String stdout, Map<String, String> env) {
```
[the remark this answers is not in the document]
Comment (nickkkccc) [reply src/test/java/org/testcontainers/containers/CartridgeContainerTestUtils.java]: Fixed

File: README.md [hunk 9]
```diff
@@ -193,6 +197,185 @@ public class SomeOtherTest {
...
+
+To set an environment variable, use the `withEnv(...)` method of testcontainers API. Full list of variables the 
+environments used in cartridge can be found here [link](https://www.tarantool.io/ru/doc/2.11/book/cartridge/cartridge_api/modules/cartridge/).
+
+***Note:*** As shown in the previous section, some build arguments are converted to environment variables and used to
+cartridge build at the image build stage.
+
+An example of how to set the `TARANTOOL_CLUSTER_COOKIE` parameter:
+
+```java
```
[the remark this answers is not in the document]
Comment (nickkkccc) [reply README.md]: Fixed

File: README.md [hunk 10]
```diff
@@ -193,6 +197,41 @@ public class SomeOtherTest {
...
+
+###### Environment variables:
+
+To set an environment variable, use the `withEnv(...)` method of testcontainers API. Full list of variables the 
+environments used in cartridge can be found here [link](https://www.tarantool.io/ru/doc/2.11/book/cartridge/cartridge_api/modules/cartridge/#cfg-opts-box-opts).
+
+***Note:*** As shown in the previous section, some build arguments are converted to environment variables and used to
+cartridge build at the image build stage.
+
+An example of how to set the `TARANTOOL_CLUSTER_COOKIE` parameter: [link](https://github.com/tarantool/testcontainers-java-tarantool/blob/8eb53c43e4c00570378de91a1269837448232de8/src/test/java/org/testcontainers/containers/TarantoolCartridgeBootstrapFromLuaWithFixedPortsTest.java#L57-L82).
```
Comment (ArtDu) [inline README.md:233]: It renders only by raw link and I updated commit to newer one

```suggestion
An example of how to set the `TARANTOOL_CLUSTER_COOKIE` parameter: https://github.com/tarantool/testcontainers-java-tarantool/blob/355d1e985bd10beca83bc7ca77f919a288709419/src/test/java/org/testcontainers/containers/TarantoolCartridgeBootstrapFromLuaWithFixedPortsTest.java#L57-L82
```

File: README.md [hunk 11]
```diff
@@ -193,6 +197,41 @@ public class SomeOtherTest {
...
+an environment variable. It is not recommended to override via the `withEnv(...)` method.
+- `TARANTOOL_LOGDIR` - the directory where log files are stored. Defaults is `/tmp/log`. Converts to an environment 
+- variable. It is not recommended to override via the `withEnv(...)` method.
+- `TARANTOOL_INSTANCES_FILE` - path to the configuration file. Defaults is `./instances.yml`. Converts to an environment
+variable. It is not recommended to override via the `withEnv(...)` method.
+- `START_DELAY` - the time after which cartridge will actually run after the container has started. Converts to an 
+environment variable. It is not recommended to override via the `withEnv(...)` method.
+
+You can set the Docker image build arguments using a map, which is passed as an input argument to the constructor when
+creating a container in Java code. See example: [link](https://github.com/tarantool/testcontainers-java-tarantool/blob/8eb53c43e4c00570378de91a1269837448232de8/src/test/java/org/testcontainers/containers/TarantoolCartridgeBootstrapFromLuaWithFixedPortsTest.java#L111-L119).
```
Comment (ArtDu) [inline README.md:223]: It renders only by raw link and I updated commit to newer one

```suggestion
creating a container in Java code. See example: https://github.com/tarantool/testcontainers-java-tarantool/blob/355d1e985bd10beca83bc7ca77f919a288709419/src/test/java/org/testcontainers/containers/TarantoolCartridgeBootstrapFromLuaWithFixedPortsTest.java#L111-L119
```

**Review by ArtDu — CHANGES_REQUESTED** [review]

**Review by ArtDu — APPROVED** [review]

**Pull request merged**